src/colorizer/shaders/colorize_RGBA8U.frag

```diff
@@ -60,6 +62,20 @@ void main() {
     gOutputColor = vec4(backgroundColor, 1.0);
     return;
   }
+  else if (int(id) - 1 == highlightedId) {
+    float thickness = 2.0;
+    float wStep = 1.0 / float(frameDims.x);
+    float hStep = 1.0 / float(frameDims.y);        
+    // sample around the pixel to see if we are on an edge
+    uint R = combineColor(texture(frame, sUv + vec2(thickness*wStep, 0)));
+    uint L = combineColor(texture(frame, sUv + vec2(-thickness*wStep, 0)));
+    uint T = combineColor(texture(frame, sUv + vec2(0, thickness*hStep)));
+    uint B = combineColor(texture(frame, sUv + vec2(0, -thickness*hStep)));
+    if (R == 0u || L == 0u || T == 0u || B == 0u) {
```

Comment:
do we need to do combinecolor if we are only testing for 0?

Reply:
you still need to check all the RGB components, though for testing for 0 you could skip the shifts that combineColor does and just say `r | g | b`.